src/multivariate/dirichlet.jl

```diff
@@ -26,10 +26,12 @@ struct Dirichlet{T<:Real,Ts<:AbstractVector{T},S<:Real} <: ContinuousMultivariat
     lmnB::S
 
     function Dirichlet{T}(alpha::AbstractVector{T}; check_args::Bool=true) where T
-        @check_args(
-            Dirichlet,
-            (alpha, all(x -> x > zero(x), alpha), "alpha must be a positive vector."),
-        )
+        if check_args
```

Comment:
This should not be needed, I think? IIRC `@check_args` already performs this check. The macro was added specifically to improve AD support.

Reply:
ah my bad I thought the check_args argument had been forgotten here

Reply:
done